### PR TITLE
Android opensles

### PR DESCRIPTION
**Pull request by mh21**

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -150,7 +150,8 @@ SRC_DIRS = \
 	src/hostapi/wdmks \
 	src/hostapi/wmme \
 	src/os/unix \
-	src/os/win
+	src/os/win \
+	src/hostapi/opensles
```
Comment (mh21) [inline Makefile.in]: move above src/os/*

File: bindings/cpp/aclocal.m4 [hunk 2]
```diff
@@ -14,8 +14,8 @@
 m4_ifndef([AC_CONFIG_MACRO_DIRS], [m4_defun([_AM_CONFIG_MACRO_DIRS], [])m4_defun([AC_CONFIG_MACRO_DIRS], [_AM_CONFIG_MACRO_DIRS($@)])])
 m4_ifndef([AC_AUTOCONF_VERSION],
   [m4_copy([m4_PACKAGE_VERSION], [AC_AUTOCONF_VERSION])])dnl
-m4_if(m4_defn([AC_AUTOCONF_VERSION]), [2.69],,
-[m4_warning([this file was generated for autoconf 2.69.
+m4_if(m4_defn([AC_AUTOCONF_VERSION]), [2.68],,
```
Comment (mh21) [inline bindings/cpp/aclocal.m4]: upgrade local autoconf version to 2.69?

File: src/common/TAGS [hunk 3]
```diff
@@ -0,0 +1,631 @@
+
```
Comment (mh21) [inline src/common/TAGS]: shouldn't be in the commit

File: src/common/pa_hostapi.h [hunk 4]
```diff
@@ -153,6 +153,14 @@ are defaulted to 1.
 #define PA_USE_ASIHPI 1
 #endif 
 
+/* TODO opensles */
```
Comment (mh21) [inline src/common/pa_hostapi.h]: why todo?

File: src/hostapi/opensles/README.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Android implementation for portaudio using opensles.
```
Comment (mh21) [inline src/hostapi/opensles/README.txt]: this could be quite a bit more text :-)

File: bindings/cpp/aclocal.m4 (hunk 2 above)
Comment (Gundersanne) [inline bindings/cpp/aclocal.m4]: just did :)

File: src/os/unix/pa_unix_hostapis.c [hunk 6]
```diff
@@ -51,6 +51,7 @@ PaError PaSGI_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex
 PaError PaAsiHpi_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 PaError PaMacCore_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 PaError PaSkeleton_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
+PaError PaOpensles_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
```
Comment (mh21) [inline src/os/unix/pa_unix_hostapis.c]: OpenSLES

**Comment by mh21** [comment]
squash all commits into one

File: src/os/unix/pa_unix_hostapis.c [hunk 7]
```diff
@@ -52,6 +52,8 @@ PaError PaAsiHpi_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiInd
 PaError PaMacCore_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 PaError PaSkeleton_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 
+PaError PaOpenSLES_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
```
Comment (mh21) [inline src/os/unix/pa_unix_hostapis.c]: empty line should be below

File: src/hostapi/opensles/pa_hostapi_opensles.c [hunk 8]
```diff
@@ -0,0 +1,965 @@
...
+ * requested that these non-binding requests be included along with the 
+ * license above.
+ */
+
+/** @file
+ @ingroup common_src
+
+ @brief opensles implementation of support for a host API.
+
+ @note IMPLEMENT ME comments are used to indicate functionality
```
Comment (mh21) [inline src/hostapi/opensles/pa_hostapi_opensles.c]: remove

File: src/hostapi/opensles/pa_hostapi_opensles.c [hunk 9]
```diff
@@ -0,0 +1,965 @@
...
+static PaTime GetStreamTime( PaStream *stream );
+static double GetStreamCpuLoad( PaStream* stream );
+static PaError ReadStream( PaStream* stream, void *buffer, unsigned long frames );
+static PaError WriteStream( PaStream* stream, const void *buffer, unsigned long frames );
+static signed long GetStreamReadAvailable( PaStream* stream );
+static signed long GetStreamWriteAvailable( PaStream* stream );
+
+
+
+/* PaSkeletonHostApiRepresentation - host api datastructure specific to this implementation */
```
Comment (mh21) [inline src/hostapi/opensles/pa_hostapi_opensles.c]: not a skeleton anymore

**Pull request closed without merging**